actions/maven/publish/action.yml

```
@@ -0,0 +1,117 @@
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+inputs:
+  provenance-download-name:
+    description: "The artifact name for the package provenance."
+    required: true
+    type: string
+  provenance-download-sha256:
+    description: "The sha256 of the package provenance artifact."
+    required: false
+    type: string
+  target-download-sha256:
+    description: "The sha256 of the target directory."
+    required: true
+    type: string
+  maven-username:
+    description: "Maven username"
+    required: false
+  maven-password:
+    description: "Maven password"
+    required: false
+  gpg-key-pass:
+    description: "gpg-key-pass"
+    required: false
+  gpg-private-key:
+    description: "gpg-key-pass"
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout the project repository
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-project-checkout@v1.8.0-rc.1
```

We can perhaps merge this after the release.
```suggestion
      uses: slsa-framework/slsa-github-generator/.github/actions/secure-project-checkout@main
```